packages/admin/blocks-admin/src/blocks/factories/createOneOfBlock.tsx

```diff
@@ -218,6 +218,27 @@ CreateOneOfBlockOptions): BlockInterface<OneOfBlockFragment, OneOfBlockState, an
             return block?.anchors?.(blockState.props) ?? [];
         },
 
+        extractTextContents: (state) => {
+            return state.attachedBlocks.reduce<string[]>((contentBlock, child) => {
+                const block = blockForType(child.type);
+
+                if (!block) {
+                    throw new Error(`No Block found for type ${child.type}`); // for TS
+                }
+                return [...contentBlock, ...(block.extractTextContents?.(child.props) ?? [])];
+            }, []);
+        },
+
+        replaceTextContents: (state, contents) => {
+            const { state: blockState, block } = getActiveBlock(state);
+
+            if (blockState === undefined) {
+                return [];
+            }
+
+            return block?.replaceTextContents?.(blockState.props, contents) ?? blockState.props;
+        },
```

Comment:
This should replace in all blocks, not only the active block. Furthermore, I believe the returned block state is incorrect